packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/add-auth-request-adaptor.ts

```diff
@@ -0,0 +1,237 @@
+  return {
+    adminQueries: !!adminQueries,
+    adminQueryGroup: adminQueries?.permissions.groupName,
+  };
+};
+
+// converts mfa config to existing format
+const mfaMap = (mfaConfig: CognitoMFAConfiguration = { mode: 'OFF' }): MfaResult => {
+  if (mfaConfig.mode === 'OFF') {
+    return {
+      mfaConfiguration: 'OFF',
+    };
+  }
+  return {
+    mfaConfiguration: mfaConfig.mode,
+    mfaTypes: mfaConfig.mfaTypes.map(type => mfaTypeMap[type]),
+    smsAuthenticationMessage: mfaConfig.smsMessage,
+  };
+};
+
+// converts password recovery config to existing format
+const passwordRecoveryMap = (pwRecoveryConfig?: CognitoPasswordRecoveryConfiguration): PasswordRecoveryResult => {
+  switch (pwRecoveryConfig?.deliveryMethod) {
+    case 'SMS':
+      return {
+        smsVerificationMessage: pwRecoveryConfig?.smsMessage,
+        autoVerifiedAttributes: ['phone_number'],
+      };
+    case 'EMAIL':
+      return {
+        emailVerificationMessage: pwRecoveryConfig?.emailMessage,
+        emailVerificationSubject: pwRecoveryConfig?.emailSubject,
+        autoVerifiedAttributes: ['email'],
+      };
+    default:
+      return {
+        autoVerifiedAttributes: [],
+      };
+  }
+};
+
+const passwordConstraintMap: Record<CognitoPasswordConstraint, PasswordPolicy> = {
+  [CognitoPasswordConstraint.REQUIRE_LOWERCASE]: 'Requires Lowercase',
+  [CognitoPasswordConstraint.REQUIRE_DIGIT]: 'Requires Numbers',
+  [CognitoPasswordConstraint.REQUIRE_SYMBOL]: 'Requires Symbols',
+  [CognitoPasswordConstraint.REQUIRE_UPPERCASE]: 'Requires Uppercase',
+};
+
+const mfaTypeMap: Record<'SMS' | 'TOTP', 'SMS Text Message' | 'TOTP'> = {
+  SMS: 'SMS Text Message',
+  TOTP: 'TOTP',
+};
+
+const signinAttributeMap: Record<CognitoUserPoolSigninMethod, UsernameAttributes> = {
+  [CognitoUserPoolSigninMethod.USERNAME]: 'username',
+  [CognitoUserPoolSigninMethod.EMAIL]: 'email',
+  [CognitoUserPoolSigninMethod.PHONE_NUMBER]: 'phone_number',
+  [CognitoUserPoolSigninMethod.EMAIL_AND_PHONE_NUMBER]: 'email, phone_number',
+};
+
+// converts a word like tHiS to This
+const toTitleCase = (word: string): string =>
+  word
+    .charAt(0)
+    .toUpperCase()
+    .concat(word.slice(1).toLowerCase());
+
+// converts A_STRING_LIKE_THIS into AStringLikeThis
+const upperSnakeCaseToUpperCamelCase = (str: string): string =>
+  str
+    .toLowerCase()
+    .replace(/(^\w|_\w)/g, group => group.toUpperCase()) // upper case first letter and any letter after an underscore
+    .replace(/_/g, ''); // remove underscores
+
+const socialFederationKeyMap = (provider: 'FACEBOOK' | 'AMAZON' | 'GOOGLE', projectType: string): string => {
+  switch (provider) {
+    case 'FACEBOOK':
+      return 'facebookAppId';
+    case 'AMAZON':
+      return 'amazonAppId';
+    case 'GOOGLE':
+      switch (projectType) {
+        case 'ios':
+          return 'googleIos';
+        case 'android':
+          return 'googleAndroid';
+        default:
```

Comment:
The project types `switch case` should handle all the known ones and throw for default, because flutter support is in dev preview and I'm pretty sure that a project type check like this in any package will not be updated as it is easy to miss.